src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonapi/PrisonApiClient.kt

```diff
@@ -108,16 +108,20 @@ class PrisonApiClient(private val prisonApiWebClient: WebClient) {
    * Returns all matching appointment types for a prison, not just video link bookings. It will however filter down to
    * the supplied location IDs where there is match.
    */
-  fun getScheduledAppointments(prisonCode: String, date: LocalDate, locationId: Collection<Long>) = prisonApiWebClient.get()
-    .uri { uriBuilder: UriBuilder ->
-      uriBuilder
-        .path("/api/schedules/{prisonCode}/appointments")
-        .queryParam("date", date)
-        .build(prisonCode)
-    }
-    .retrieve()
-    .bodyToMono(typeReference<List<ScheduledAppointment>>())
-    .block() ?: emptyList()
+  fun getScheduledAppointments(prisonCode: String, date: LocalDate, locationIds: Collection<Long>) = run {
+    (
+      prisonApiWebClient.get()
+        .uri { uriBuilder: UriBuilder ->
+          uriBuilder
+            .path("/api/schedules/{prisonCode}/appointments")
+            .queryParam("date", date)
+            .build(prisonCode)
+        }
+        .retrieve()
+        .bodyToMono(typeReference<List<ScheduledAppointment>>())
+        .block() ?: emptyList()
+      ).filter { appointment -> locationIds.isEmpty() || appointment.locationId in locationIds }
```

Comment:
This method was not filtering out locations.  This was impacting the room look up endpoint, it was including non-video link locations which broke the process.  I noticed this during testing in dev.